projects/angular/src/data/datagrid/render/main-renderer.ts

```diff
@@ -213,6 +218,47 @@ export class DatagridMainRenderer implements AfterContentInit, AfterViewInit, Af
     });
   }
 
+  private columnStateChanged(state) {
+    const columnIndex = state.columnIndex;
+    if (state.changes && state.changes.length) {
+      state.changes.forEach(change => {
+        switch (change) {
+          case DatagridColumnChanges.WIDTH:
+            this.headers.get(columnIndex).setWidth(state);
+            this.rows.forEach(row => {
+              if (row.cells && row.cells.length) {
+                row.cells.get(columnIndex).setWidth(state);
+              }
+            });
+            break;
+          case DatagridColumnChanges.HIDDEN:
+            this.headers.get(columnIndex).setHidden(state);
+            this.rows.forEach(row => {
+              if (row.cells && row.cells.length) {
+                row.cells.get(columnIndex).setHidden(state);
+              }
+            });
+            this.keyNavigation.resetKeyGrid();
+            break;
+          default:
+            break;
+        }
+      });
+    }
+  }
+
+  private listenForColumnChanges() {
+    this.columnsService.columns.forEach((column, index) => {
+      this.columnsService.emitStateChange(column, { changes: [], columnIndex: index } as ColumnStateDiff);
```

Comment:
```suggestion
      this.columnsService.emitStateChange(column, { changes: [], columnIndex: index });
```

This type assertion is unnecessary.

Reply:
Thanks, fixed in [cb1c20d](https://github.com/vmware-clarity/ng-clarity/pull/1346/commits/cb1c20da71a5cca6b781799d9b36f8d5fb2b80c6)